### PR TITLE
[fix] Fix crash when no diff between old and new target-branch

### LINKED ISSUE #108: Make sure sanitizing from solutions branch doesnt crash with 0 diff
I believe there was a bug before when we tested sanitizing to the master branch. If there was no difference between master and the sanitized solutions sanitizer would crash/break/complain. There was no issue created for this and therefore we should test it to see if it is still a problem.

### PR DESCRIPTION
**Pull request by tohanss**
Fix #108

### CODE REVIEW

File: repobee_sanitizer/_sanitize_repo.py [hunk 1]
```diff
@@ -136,7 +140,11 @@ def _git_commit_on_branch(repo_root: pathlib.Path, target_branch: str):
     repo = git.Repo(str(repo_root))
     repo.git.symbolic_ref("HEAD", f"refs/heads/{target_branch}")
     repo.git.add(".", "--force")
-    repo.git.commit("-m", "'Sanitize files'")
+    try:
+        repo.git.commit("-m", "'Sanitize files'")
+    except git.GitCommandError as exc:
+        if "nothing to commit, working tree clean" in exc.__str__():
```
Comment (slarse) [inline repobee_sanitizer/_sanitize_repo.py:146]: In general, we don't invoke dunder (double underscore) methods directly.

```suggestion
        if "nothing to commit, working tree clean" in str(exc):
```

File: repobee_sanitizer/_sanitize_repo.py [hunk 2]
```diff
@@ -136,7 +140,11 @@ def _git_commit_on_branch(repo_root: pathlib.Path, target_branch: str):
...
+            raise EmptyCommitError()
```
Comment (slarse) [inline repobee_sanitizer/_sanitize_repo.py:147]: Always do `raise from` to include the whole stacktrace.

```suggestion
            raise EmptyCommitError() from exc
```

File: repobee_sanitizer/_sanitize_repo.py (hunk 1 above)
Comment (slarse) [inline repobee_sanitizer/_sanitize_repo.py:146]: I'm thinking that perhaps we don't need this `if` statement. We currently aren't aware that there can be any other form of failure here, so we should probably just put in an assert instead.
```suggestion
        assert "nothing to commit, working tree clean" in str(exc)
```

Thoughts?

Comment (slarse) [reply repobee_sanitizer/_sanitize_repo.py:146]: Now I'm thinking that perhaps we don't need this `if` statement. We currently aren't aware that there can be any other form of failure here, so we should probably just put in an assert instead.
```suggestion
        assert "nothing to commit, working tree clean" in str(exc)
```

Thoughts?

Comment (tohanss) [reply repobee_sanitizer/_sanitize_repo.py]: I have no idea why str() flew above my head for that one, but its fixed now :)

**Pull request merged**